src/components/share-drawer/ShareDrawer.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback, useContext } from 'react'
+
+import Clipboard from '@react-native-clipboard/clipboard'
+import { CommonState } from 'audius-client/src/common/store'
+import { getAccountUser } from 'audius-client/src/common/store/account/selectors'
+import { getUser } from 'audius-client/src/common/store/cache/users/selectors'
+import {
+  getModalVisibility,
+  setVisibility
+} from 'audius-client/src/common/store/ui/modals/slice'
+import {
+  getSource,
+  getTrack
+} from 'audius-client/src/common/store/ui/share-modal/selectors'
+import { requestOpen as requestOpenTikTokModal } from 'audius-client/src/common/store/ui/share-sound-to-tiktok-modal/slice'
+import { Linking, StyleSheet, View } from 'react-native'
+
+import IconLink from 'app/assets/images/iconLink.svg'
+import IconShare from 'app/assets/images/iconShare.svg'
+import IconTikTok from 'app/assets/images/iconTikTok.svg'
+import IconTikTokInverted from 'app/assets/images/iconTikTokInverted.svg'
+import IconTwitterBird from 'app/assets/images/iconTwitterBird.svg'
+import Text from 'app/components/text'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { useThemedStyles } from 'app/hooks/useThemedStyles'
+import { getTrackRoute } from 'app/utils/routes'
+import {
+  Theme,
+  ThemeColors,
+  useThemeColors,
+  useThemeVariant
+} from 'app/utils/theme'
+
+import ActionDrawer from '../action-drawer'
+import { ToastContext } from '../toast/ToastContext'
+
+import { getTwitterLink } from './utils'
+
+const messages = {
+  modalTitle: 'Share Track',
+  twitter: 'Share to Twitter',
+  tikTok: 'Share Sound to TikTok',
+  copyLink: 'Copy Track to Link'
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    shareToTwitterAction: {
+      color: '#1BA1F1'
```

Comment:
It might be worth adding this color to our themes

Reply:
yes agreed, that was on a list to bring up! so would this go in the `defaultTheme` object?

Reply:
Yeah, you will need to put it in all 3 themes. It could be nice to have some static colors outside of themes though, such as `staticWhite`. But I kinda like having each theme fully declarative

Reply:
Also I want to move the theme definitions out of the utils folder at some point...